perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,161 @@
+
+    Args:
+      query: Query that will be executed in BigQuery.
+
+    Raises:
+      Aborts the program if error is encountered while executing th query.
+    """
+    job = self.client.query(query)
+    job.result()
+    if job.errors:
+      for error in job.errors:
+        print(f"Error message: {error['message']}")
+      sys.exit(1)
+    return
+
+  def setup_bigquery(self):
+
+    """Creates the experiment configuration table to store the configuration details and
+       creates the list_metrics, read_write_fio_metrics and read_write_vm_metrics tables
+       to store the metrics data if they don't already exist in the dataset
+    """
+
+    # Query for creating experiment_configuration table if it does not exist
+    query_create_table_experiment_configuration = """
+      CREATE TABLE IF NOT EXISTS {}.{}.{}(
+        configuration_id INT64,
+        configuration_name STRING,
+        gcsfuse_flags STRING,
+        branch STRING,
+        end_date TIMESTAMP,
+        PRIMARY KEY (configuration_id) NOT ENFORCED
+      ) OPTIONS (description = 'Table for storing Job Configurations and respective VM instance name on which the job was run');
+    """.format(self.project_id, self.dataset_id, BigQuery.configuration_table_id)
+
+    # Query for creating fio_metrics table
+    query_create_table_fio_metrics = """
+      CREATE TABLE IF NOT EXISTS {}.{}.{}(
+        configuration_id INT64, 
+        start_time_build TIMESTAMP,
+        test_type STRING, 
+        num_threads INT64, 
+        file_size_kb INT64, 
+        block_size_kb INT64,
+        start_time INT64, 
+        end_time INT64, 
+        iops FLOAT64, 
+        bandwidth_bytes_per_sec INT64, 
+        io_bytes INT64, 
+        min_latency FLOAT64, 
+        max_latency FLOAT64, 
+        mean_latency FLOAT64, 
+        percentile_latency_20 FLOAT64, 
+        percentile_latency_50 FLOAT64, 
+        percentile_latency_90 FLOAT64, 
+        percentile_latency_95 FLOAT64, 
+        FOREIGN KEY(configuration_id) REFERENCES {}.{} (configuration_id) NOT ENFORCED
+      ) OPTIONS (description = 'Table for storing FIO metrics extracted from experiments.');
+    """.format(self.project_id, self.dataset_id, BigQuery.fio_table_id, self.dataset_id, BigQuery.configuration_table_id)
+
+    # Query for creating vm_metrics table
+    query_create_table_vm_metrics = """
+      CREATE TABLE IF NOT EXISTS {}.{}.{}(
+        configuration_id INT64, 
+        start_time_build TIMESTAMP,
+        end_time INT64, 
+        cpu_utilization_peak_percentage FLOAT64, 
+        cpu_utilization_mean_percentage FLOAT64, 
+        received_bytes_peak_per_sec FLOAT64, 
+        received_bytes_mean_per_sec FLOAT64, 
+        read_bytes_count INT64,
+        ops_error_count INT64, 
+        ops_mean_latency_sec FLOAT64, 
+        sent_bytes_peak_per_sec FLOAT64, 
+        sent_bytes_mean_per_sec FLOAT64, 
+        sent_bytes_count INT64,
+        memory_utilization_ram FLOAT64,
```

Comment:
let's remove memory_utilization_ram & memory_utilization_disk_tempdir for now. Add them when we start exporting them.